src/main/kotlin/no/nav/helse/domene/person/PersonService.kt

```diff
@@ -40,4 +45,35 @@ class PersonService(private val personClient: PersonClient) {
             }.map {
                 GeografiskTilknytningMapper.tilGeografiskTilknytning(it)
             }
+
+    fun barn(aktørId: AktørId) =
+        personClient.familierelasjoner(aktørId).toEither { error ->
+            log.error("Feil ved oppslag på familierelasjoner", error)
+            when (error) {
+                is HentPersonPersonIkkeFunnet -> Feilårsak.IkkeFunnet
+                is HentPersonSikkerhetsbegrensning -> Feilårsak.FeilFraTjeneste
+                else -> Feilårsak.FeilFraTjeneste
+            }
+        }.map { familierelasjoner ->
+            /*
+                "tilPerson" som blir returnert som en del av Familierelasjon inneholder kun aktørId og navn.
+                Må gjøre oppslag på hvert enkelt barn for å kunne avgjøre status og disresjonskode
+             */
+            val barnAktørIder =  familierelasjoner
+                    .filter { "BARN" == it.tilRolle.value }
+                    .filter { aktiveEndringstyper.contains(it.endringstype) }
+                    .map { AktørId((it.tilPerson.aktoer as AktoerId).aktoerId) }
+
+            log.trace("Slår opp info på '${barnAktørIder.size}' barn.")
+
+            val barn = mutableListOf<Person>()
+            barnAktørIder.forEach { barnAktørId ->
+                val oppslagResult = personInfo(barnAktørId)
+                when (oppslagResult) {
+                    is Either.Right -> barn.add(oppslagResult.b)
+                    is Either.Left -> log.error("Feil ved oppslag på barn med AktørID '$barnAktørId'")
+                }
+            }
+            barn.toList()
```

Comment:
i sparkel så logges det meste på info. Ting som logges på `error` bør vurderes om skal ende i feil. Dersom det ikke skal ende i feil, så er `info` mer riktig.

I stedet for `}.map { familierelasjoner ->` så burde man heller bruke `flatMap`. Da forventes det at en ny either returneres, noe vi kunne gjort etter `personInfo`-oppslaget på barnet:

```
}.flatMap { familierelasjoner ->
    familierelasjoner
                    .filter { "BARN" == it.tilRolle.value }
                    .filter { aktiveEndringstyper.contains(it.endringstype) }
                    .map { AktørId((it.tilPerson.aktoer as AktoerId).aktoerId) }
                    .map { barnAktørId ->
                        personInfo(barnAktørId)
                    }
}
```

Reply:
Hmm, ikke helt med på denne kjenner jeg. Ender jo opp med en Liste av Either<Feilmelding, Barn>, er det noe mer smidig måte å løse dette på med arrow?

Reply:
det er en funksjon som heter `sequenceU()` som gjør om en `List<Either<A, B>>` til `Either<A, List<B>>`